### PR TITLE
feat: Implement video frame resize callback

### PR DESCRIPTION
**Pull request by kannan-xiao4**
Issue https://github.com/Unity-Technologies/com.unity.webrtc/issues/507

### CODE REVIEW

**Comment by doctorpangloss** [comment]
Does it make sense to allocate the render texture on the plugin side? It would work cleanly even if you have to `Release` and `Create` because the Unity runtime reference to the render texture would remain the same.

File: Runtime/Scripts/VideoStreamTrack.cs [hunk 1]
```diff
@@ -184,6 +194,19 @@ public override void Dispose()
             }
             base.Dispose();
         }
+
+        internal void OnVideoFrameResize(int width, int height)
```
Comment (kannan-xiao4) [inline Runtime/Scripts/VideoStreamTrack.cs:198]: @doctorpangloss 
Is your concern that this method indicate that you are creating a RenderTexture?

Comment (doctorpangloss) [reply Runtime/Scripts/VideoStreamTrack.cs:198]: yes, this is going to leak textures if the width and height change.

Comment (kannan-xiao4) [reply Runtime/Scripts/VideoStreamTrack.cs:198]: thanks
I understand and fixed in this commit ad70ecc

File: Runtime/Scripts/VideoStreamTrack.cs [hunk 2]
```diff
@@ -184,6 +194,40 @@ public override void Dispose()
...
+            if (m_sourceTexture != null && (m_sourceTexture.width == width && m_sourceTexture.height == height))
+            {
+                return;
+            }
+
+            if (m_sourceTexture != null && m_sourceTexture is RenderTexture source)
+            {
+                source.Release();
+                source.width = width;
+                source.height = width;
```
Comment (karasusan) [inline Runtime/Scripts/VideoStreamTrack.cs:209]:
```suggestion
                source.height = height;
```

File: Runtime/Scripts/VideoStreamTrack.cs [hunk 3]
```diff
@@ -184,6 +194,40 @@ public override void Dispose()
...
+            else
+            {
+                m_sourceTexture = CreateRenderTexture(width, height);
+            }
+
+            if (m_destTexture != null)
+            {
+                m_destTexture.Release();
+                m_destTexture.width = width;
+                m_destTexture.height = width;
```
Comment (karasusan) [inline Runtime/Scripts/VideoStreamTrack.cs:221]:
```suggestion
                m_destTexture.height = height;
```

**Review by karasusan — APPROVED** [review]

**Pull request merged**